src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala

```diff
@@ -25,9 +33,54 @@ object CakeBuildInfoPlugin extends AutoPlugin {
         Try("git rev-parse --verify HEAD".!! dropRight 1) getOrElse "n/a"
       )
     ),
-    buildInfoPackage := s"${organization.value}.${(name in ThisBuild).value}.build",
+    buildInfoPackage :=
+      s"${organization.value}.${(name in ThisBuild).value}.build",
     buildInfoOptions += BuildInfoOption.BuildTime,
     buildInfoOptions += BuildInfoOption.ToJson
   )
 
 }
+
+/**
+  * Build keys that will be auto-imported when this plugin is enabled.
+  */
+object CakeBuildInfoKeys {
+  /**
+    * Map holding generic set of labelled values. These are use to label jar
+    * manifests and Docker containers.
+    */
+  val generalInfo: Map[String, String] = Map(
+    "buildDatetime" -> buildDatetime,
```

Comment:
Yes, this was intentional. We only need to construct this data once, typically when we publish artefacts (though based on a suggestion below, we might also need to do this for build info as well?). So, we either achieve this via a lazy val or a val - I simply preferred the latter over the former.